components/function-controller/internal/controllers/serverless/service.go

```diff
@@ -92,3 +85,29 @@ func (r *FunctionReconciler) updateService(ctx context.Context, log logr.Logger,
 		Message:            fmt.Sprintf("Service %s updated", svc.GetName()),
 	})
 }
+
+func (r *FunctionReconciler) deleteExcessServices(ctx context.Context, instance *serverlessv1alpha1.Function, log logr.Logger, services []corev1.Service) (ctrl.Result, error) {
+	// services do not support deletecollection
+	// you can check this by `kubectl api-resources -o wide | grep services`
+	// also https://github.com/kubernetes/kubernetes/issues/68468#issuecomment-419981870
+
+	log.Info("Deleting excess Services")
+
+	for i, _ := range services {
+		svc := services[i]
+		if svc.GetName() == instance.GetName() {
+			continue
+		}
+
+		log.Info(fmt.Sprintf("Deleting Service %s", svc.GetName()))
+
+		err := r.client.Delete(ctx, &services[i])
```

Comment:
`r.client.Delete(ctx, &svc)`

Reply:
&svc adress is always the same, as we do a for loop here and svc gets always the same address, it's safer to use &services[i] as it'll get different pointer every time